src/App.js

```diff
@@ -0,0 +1,242 @@
+    console.log(error);
+    throw new Error(`Error when liking card ${id}`);
+  }
+};
+
+function App() {
+  const [boards, setBoards] = useState([]); // list of all the board dicts
+  const [boardOption, setBoardOption] = useState("Choose a Board");
+  const [chosenBoardData, setChosenBoardData] = useState({ cards: [] });
+  const [cardOrder, setSortOrder] = useState("");
+  const [cardSort, setSortType] = useState("");
+  const [sortedData, setSortedData] = useState([]);
+  const [isBoardFormVisible, setIsBoardFormVisible] = useState(false);
+
+  const showChosenBoard = (boardTitle) => {
+    setBoardOption(boardTitle);
+  };
+
+  // order is ascending or descending, type is which type of sort
+  const updateSortOrder = (order) => {
+    setSortOrder(order);
+  };
+
+  const updateSortType = (type) => {
+    setSortType(type);
+  };
+
+  const getBoardListDropdown = () => {
+    axios
+      .get(`${kBaseUrl}/boards`)
+      .then((response) => {
+        setBoards(response.data);
+      })
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Unable to get board options");
+      });
+  };
+
+  const createNewBoard = (newBoard) => {
+    axios
+      .post(`${kBaseUrl}/boards`, newBoard)
+      .then(() => getBoardListDropdown())
+      .then(() => setBoardOption(newBoard.title))
+      .then(() => setChosenBoardData({ cards: [] }))
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Couldn't create a new board.");
+      });
+  };
+
+  const toggleNewBoardForm = () => {
+    setIsBoardFormVisible(!isBoardFormVisible);
+  };
+
+  useEffect(() => {
+    // initially loads boards
+    getBoardListDropdown();
+  }, []);
+
+  useEffect(() => {
+    if (cardOrder && cardSort) {
+      let sortedCardData;
+      if (cardOrder === "Ascending") {
+        if (cardSort === "ID") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.id > b.id ? 1 : -1
+          );
+        } else if (cardSort === "Likes") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.likes > b.likes ? 1 : -1
+          );
+        } else if (cardSort === "Alphabetically") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.message.toLowerCase() > b.message.toLowerCase() ? 1 : -1
+          );
+        }
+      } else if (cardOrder === "Descending") {
+        if (cardSort === "ID") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.id < b.id ? 1 : -1
+          );
+        } else if (cardSort === "Likes") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.likes < b.likes ? 1 : -1
+          );
+        } else if (cardSort === "Alphabetically") {
+          sortedCardData = chosenBoardData.cards.sort((a, b) =>
+            a.message.toLowerCase() < b.message.toLowerCase() ? 1 : -1
+          );
+        }
+      }
+      setSortedData([...sortedCardData]);
```

Comment:
When you have long if/else trees whose conditions are looking to match something particular, a `switch` statement can be a good replacement: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch